lib/http2.c

```diff
@@ -394,6 +395,10 @@ static ssize_t send_callback(nghttp2_session *h2,
   written = ((Curl_send*)c->send_underlying)(data, FIRSTSOCKET,
                                              mem, length, &result);
 
+  if (written >= 0 && (size_t) written < length) {
+    H2BUGF(infof(data, "[SUPER-6847] we don't have implementation to handle this case"));
+  }
+
```

Comment:
In the PR, when written is smaller than actual buffer, then it returns `CURL_AGAIN`. We don't have buffer to keep them until next request, I couldn't port that. But this code is for sending data, used by PUT, POST, etc. This might not relate to the issue directly.